### DISCUSSION TITLE
Smart Playlist sort by multiple fields

### DISCUSSION THREAD

**Discussion by ArturRuta** [navidrome/navidrome#3132] [Q&A]
I'm trying to create a smart playlist where the output is sorted by multiple fields.
More specifically select the recordings of an artist and I'm trying to Sort by Album and tracknumber which is something i think mades a lot of sense.
Have not found any sample on how to sort by multiple fields. 
Just sepparating the fields with comma does not work (the playlist is generate, just not sorted in any way)

Have not found anything on this topic.

Can anyone tell if it feasible and how?

A working sample follows
{
  "name": "Howard Shore",
  "comment": "Artist - Howard Shore",
  "all": [
    {"is": {"artist": "Howard Shore"}}
  ],
  "sort": "album",
  "order": "asc"
}

A sample that does not sort follows

{
  "name": "Howard Shore",
  "comment": "Artist - Howard Shore",
  "all": [
    {"is": {"artist": "Howard Shore"}}
  ],
  "sort": "album,tracknumber",
  "order": "asc"
}

Thanks a lot in advance.

**Comment by deluan** [comment]
This is currently not possible. But I'll try to implement as part of the BFR #2709

**Comment by lowgos** [comment]
Hi, sorry for the necro, is this supported?
[The documentation says no](https://www.navidrome.org/docs/usage/smartplaylists/#sorting-by-multiple-fields)

I just wanted to check out if things changed.

**Reply by deluan** [reply]
Just merged it: https://github.com/navidrome/navidrome/pull/4214

**Reply by lowgos** [reply]
Damn that was quick! Thank you so much for the trouble of doing this in less than an hour!
I hope you are not too overworked.

Thanks again for this amazing project!

**Reply by deluan** [reply]
I had some old code for this, in an old git stash. When I saw your message I just decided to finish it up. :)
